File: lib_common/src/main/java/com/example/common/widget/textview/TimeTextView.java
package com.example.common.widget.textview;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.Gravity;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.example.base.utils.TimerHelper;

import java.text.MessageFormat;

/**
 * Created by wangyanbin
 * 倒计时textview
 * 配置enable的xml和默認text文案即可
 */
@SuppressLint("AppCompatCustomView")
public class TimeTextView extends TextView {

    public TimeTextView(Context context) {
        super(context);
        initialize();
    }

    public TimeTextView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        initialize();
    }

    public TimeTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initialize();
    }

    //公共属性，可在此配置
    private void initialize() {
        setGravity(Gravity.CENTER);
    }

    public void countDown() {
        countDown(60);
    }

    public void countDown(long second) {
        TimerHelper.startDownTask(aLong -> {
            setEnabled(false);
            setText(MessageFormat.format("已发送{0}S", aLong));
            return null;
        }, () -> {
            setEnabled(true);
            setText("重发验证码");
            return null;
        }, second);
    }

    @Override
    protected void onDetachedFromWindow() {
        super.onDetachedFromWindow();
        TimerHelper.stopDownTask();
    }

}

File: lib_common/src/main/java/com/example/common/widget/textview/TimeTextView.kt
package com.example.common.widget.textview

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.widget.TextView
import com.example.base.utils.TimerUtil

/**
 * Created by wangyanbin
 * 倒计时textview
 * 配置enable的xml和默認text文案即可
 */
@SuppressLint("AppCompatCustomView", "SetTextI18n")
class TimeTextView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : TextView(context, attrs, defStyleAttr) {
    private var tag = ""

    init {
        gravity = Gravity.CENTER
    }

    fun start(tag: String = javaClass.simpleName, time: Long = 60) {
        this.tag = tag
        TimerUtil.startCountDown(tag, { second: Long? ->
            isEnabled = false
            text = "已发送${second}S"
        }, {
            isEnabled = true
            text = "重发验证码"
        }, time)
    }

    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()
        TimerUtil.stopCountDown(tag)
    }

}